commit 062c8a4ff40fedf82c1ec177a63b06c41801c2e9
Author: Harald Hoyer <harald@redhat.com>
Date:   Sat Feb 6 15:44:42 2016 +0100

    tpm_eventlog.c: fix binary_bios_measurements
    
    commit 186d124f07da193a8f47e491af85cb695d415f2f upstream.
    
    The commit 0cc698af36ff ("vTPM: support little endian guests") copied
    the event, but without the event data, did an endian conversion on the
    size and tried to output the event data from the copied version, which
    has only have one byte of the data, resulting in garbage event data.
    
    [jarkko.sakkinen@linux.intel.com: fixed minor coding style issues and
     renamed the local variable tempPtr as temp_ptr now that there is an
     excuse to do this.]
    
    Signed-off-by: Harald Hoyer <harald@redhat.com>
    Fixes: 0cc698af36ff ("vTPM: support little endian guests")
    Reviewed-by: Jarkko Sakkinen <jarkko.sakkinen@linux.intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/char/tpm/tpm_eventlog.c b/drivers/char/tpm/tpm_eventlog.c
index bd72fb0..4e6940a 100644
--- a/drivers/char/tpm/tpm_eventlog.c
+++ b/drivers/char/tpm/tpm_eventlog.c
@@ -232,7 +232,7 @@ static int tpm_binary_bios_measurements_show(struct seq_file *m, void *v)
 {
 	struct tcpa_event *event = v;
 	struct tcpa_event temp_event;
-	char *tempPtr;
+	char *temp_ptr;
 	int i;
 
 	memcpy(&temp_event, event, sizeof(struct tcpa_event));
@@ -242,10 +242,16 @@ static int tpm_binary_bios_measurements_show(struct seq_file *m, void *v)
 	temp_event.event_type = do_endian_conversion(event->event_type);
 	temp_event.event_size = do_endian_conversion(event->event_size);
 
-	tempPtr = (char *)&temp_event;
+	temp_ptr = (char *) &temp_event;
 
-	for (i = 0; i < sizeof(struct tcpa_event) + temp_event.event_size; i++)
-		seq_putc(m, tempPtr[i]);
+	for (i = 0; i < (sizeof(struct tcpa_event) - 1) ; i++)
+		seq_putc(m, temp_ptr[i]);
+
+	temp_ptr = (char *) v;
+
+	for (i = (sizeof(struct tcpa_event) - 1);
+	     i < (sizeof(struct tcpa_event) + temp_event.event_size); i++)
+		seq_putc(m, temp_ptr[i]);
 
 	return 0;
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1324_062c8a4ff40fedf82c1ec177a63b06c41801c2e9.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1324_062c8a4ff40fedf82c1ec177a63b06c41801c2e9.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
